[Grid Lite] IgxGridLiteColumnConfiguration requires explicit generic type and cannot be inferred

## Description  
`IgxGridLiteColumnConfiguration` is defined as a generic type, requiring an explicit model type (e.g. `IgxGridLiteColumnConfiguration<ProductInfo>`).
Using the type without a generic argument (`IgxGridLiteColumnConfiguration[]`) results in a TypeScript error, even when the grid data and columns clearly target a single known data model.

 * igniteui-angular version: ^21.1.0-rc.1
 * browser: Chrome

## Steps to reproduce  

1. Define your columns list like this
```ts
  public columns: IgxGridLiteColumnConfiguration[] = [
        { field: 'id', header: 'ID', hidden: true, resizable: true, sortable: false, filterable: false, dataType: 'string'},
        { field: 'name', header: 'Product Name', hidden: false, resizable: true, sortable: false, filterable: false, dataType: 'string'},
        { field: 'price', header: 'Price', hidden: false, resizable: true, sortable: false, filterable: false, dataType: 'number' },
        { field: 'sold', header: 'Units sold', hidden: false, resizable: true, sortable: false, filterable: false, dataType: 'number' },
        { field: 'total', header: 'Total sold', hidden: false, resizable: true, sortable: false, filterable: false, dataType: 'number' },
        { field: 'rating', header: 'Customer rating', hidden: false, resizable: true, sortable: false, filterable: false, dataType: 'number' }
    ];
```
2. Use the respective properties in your grid
```html
 <igx-grid-lite-column [field]="col.field" [header]="col.header" [hidden]="col.hidden"
            [dataType]="col.dataType"> </igx-grid-lite-column>
```
3. Save and see the console. The following issue appears:
```bash
[ERROR] TS2322: Type 'string | number | symbol' is not assignable to type 'Keys<{ [x: string]: {}; }>'.
  Type 'symbol' is not assignable to type 'Keys<{ [x: string]: {}; }>'. [plugin angular-compiler]
```
4. When we give the columns list this type it works
```ts
columns: IgxGridLiteColumnConfiguration<ProductInfo>[]
```

## Result  
We are not able to use IgxGridLiteColumnConfiguration  without specifying  a generic type argument

## Expected result  
We should be able to use IgxGridLiteColumnConfiguration without specifying a generic type argument

--- a/projects/igniteui-angular/grids/lite/src/grid-lite-column.component.ts
+++ b/projects/igniteui-angular/grids/lite/src/grid-lite-column.component.ts
@@ -76,7 +76,7 @@ export class IgxGridLiteColumnComponent<T extends object = any> {
     //#region Inputs
 
     /** The field from the data for this column. */
-    public readonly field = input<Keys<T>>();
+    public readonly field = input<NoInfer<Keys<T>>>();
 
     /** The data type of the column's values. */
     public readonly dataType = input<DataType>('string');

--- a/src/app/grid-lite/grid-lite.sample.html
+++ b/src/app/grid-lite/grid-lite.sample.html
@@ -35,14 +35,26 @@
     </ng-template>
   </igx-grid-lite-column>
 
-  <igx-grid-lite-column
-    field="age"
-    dataType="number"
-    header="Age"
-    filterable
-    resizable
-    sortable
-  ></igx-grid-lite-column>
+  @for(column of columns; track column.field) {
+    <igx-grid-lite-column
+        [field]="column.field"
+        [header]="column.header"
+        [dataType]="column.dataType"
+        filterable
+        resizable
+        sortable
+    >
+    </igx-grid-lite-column>
+  }
+  @for(column of typedColumns; track column.field) {
+    <igx-grid-lite-column
+        [field]="column.field"
+        [header]="column.header"
+        [dataType]="column.dataType"
+    >
+    </igx-grid-lite-column>
+  }
+
   <igx-grid-lite-column field="active" dataType="boolean" header="Active">
     <ng-template
       igxGridLiteCell

--- a/src/app/grid-lite/grid-lite.sample.ts
+++ b/src/app/grid-lite/grid-lite.sample.ts
@@ -1,6 +1,6 @@
 import { Component, CUSTOM_ELEMENTS_SCHEMA, inject, viewChild } from '@angular/core';
-import { IgxGridLiteCellTemplateDirective, IgxGridLiteColumnComponent, IgxGridLiteComponent, IgxGridLiteFilteringExpression, IgxGridLiteHeaderTemplateDirective, IgxGridLiteSortingExpression, IgxGridLiteSortingOptions } from "igniteui-angular/grids/lite";
-import { GridLiteDataService } from './data.service';
+import { IgxGridLiteCellTemplateDirective, IgxGridLiteColumnComponent, IgxGridLiteColumnConfiguration, IgxGridLiteComponent, IgxGridLiteFilteringExpression, IgxGridLiteHeaderTemplateDirective, IgxGridLiteSortingExpression, IgxGridLiteSortingOptions } from "igniteui-angular/grids/lite";
+import { GridLiteDataService, type User } from './data.service';
 import { IgxCheckboxComponent } from 'igniteui-angular';
 @Component({
     selector: 'app-grid-lite-sample',
@@ -11,8 +11,16 @@ import { IgxCheckboxComponent } from 'igniteui-angular';
 })
 export class GridLiteSampleComponent {
     protected grid = viewChild<IgxGridLiteComponent<any>>('grid');
-    protected data = [];
+    protected data: User[] = [];
     private dataService = inject(GridLiteDataService);
+
+    public columns: IgxGridLiteColumnConfiguration[] = [
+        { field: 'age', header: 'Age', dataType: 'number' },
+    ];
+
+    public typedColumns: IgxGridLiteColumnConfiguration<User>[] = [
+        { field: 'email', header: 'Email', dataType: 'string' },
+    ];
 
     protected sortingExpressions: IgxGridLiteSortingExpression[] = [
         {

--- a/src/app/grid-lite/grid-lite.sample.ts
+++ b/src/app/grid-lite/grid-lite.sample.ts
@@ -1,12 +1,21 @@
-import { Component, CUSTOM_ELEMENTS_SCHEMA, inject, viewChild } from '@angular/core';
-import { IgxGridLiteCellTemplateDirective, IgxGridLiteColumnComponent, IgxGridLiteColumnConfiguration, IgxGridLiteComponent, IgxGridLiteFilteringExpression, IgxGridLiteHeaderTemplateDirective, IgxGridLiteSortingExpression, IgxGridLiteSortingOptions } from "igniteui-angular/grids/lite";
-import { GridLiteDataService, type User } from './data.service';
+import { Component, inject, viewChild } from '@angular/core';
 import { IgxCheckboxComponent } from 'igniteui-angular';
+import {
+    IgxGridLiteCellTemplateDirective,
+    IgxGridLiteColumnComponent,
+    type IgxGridLiteColumnConfiguration,
+    IgxGridLiteComponent,
+    type IgxGridLiteFilteringExpression,
+    IgxGridLiteHeaderTemplateDirective,
+    type IgxGridLiteSortingExpression,
+    type IgxGridLiteSortingOptions,
+} from "igniteui-angular/grids/lite";
+import { GridLiteDataService, type User } from './data.service';
+
 @Component({
     selector: 'app-grid-lite-sample',
     templateUrl: 'grid-lite.sample.html',
     styleUrls: ['grid-lite.sample.scss'],
-    schemas: [CUSTOM_ELEMENTS_SCHEMA],
     imports: [IgxCheckboxComponent, IgxGridLiteComponent, IgxGridLiteColumnComponent, IgxGridLiteHeaderTemplateDirective, IgxGridLiteCellTemplateDirective]
 })
 export class GridLiteSampleComponent {

